Building with cmake

hryniuk opened the pull request:
Hello :-)! I've added these `CMakeLists.txt`s and a note in `README.md`. Now it builds on Linux without any problem. When this PR will be accepted, I'll add next one with Travis CI configuration.

I didn't manage to use these `CMakeLists.txt` files in CLion under Windows, but I think it's matter of a proper IDE configuration (path to Boost and path to SFML). You can try ;-).

Let me know, if I made any mistake in `README.md`, please.

serin-delaunay commented:
This looks good so far. There are some more libraries to add; they may have been undetected because I don't include them via NuGet:
* `rapidjson` is required
* `libnoise` will become a dependency once #15 is implemented (which is my top priority at the moment)

I'll add a note to the end of README.md describing the usage of Wangview.

hryniuk commented:
Oh, ok, good to know, I'll add them.

hryniuk commented:
Finally I've got it 😄! I've:
* added **RapidJSON** library to `CMakeLists.txt` for **Wangscape**
* added `CMakeLists.txt` for **WangscapeTest** and proper dependencies
* made static library from common **Wangscape** and **WangscapeTest** code. It links with Wangscape and WangscapeTest (no need to compile it twice)
* and, of course, updated `README.md`.

serin-delaunay commented:
Great work! Thanks again :)

hryniuk commented:
Great! Now I can dive into your code if you don't mind ;).

serin-delaunay commented:
Go ahead! :D

hryniuk commented on README.md:
```diff
@@ -20,3 +20,51 @@ Rather than matching graphical tiles with the representation of logical tiles on
 * In future this will be much more customisable, using Perlin noise to generate noisy gradients and boundaries betwen corners.
 * Currently all tile generation is done through a command-line application configured using a JSON file. In the future, this should be migrated to a GUI interface which allows the user to alter tile generation parameters and immediately see the range of possible results.
+
+# How can it build?
```
Oh, it should be probably more like "How can I build the Wangscape?", didn't notice.

serin-delaunay replied:
"How can I build Wangscape?" would be perfect :)
